src/backend/distributed/metadata/node_metadata.c

```diff
@@ -641,6 +643,16 @@ ActivateNode(char *nodeName, int nodePort)
 
 	WorkerNode *newWorkerNode = SetNodeState(nodeName, nodePort, isActive);
 
+	/*
+	 * Coordinator has always the authoritative metadata, reflect this
+	 * fact in the pg_dist_node.
+	 */
+	if (newWorkerNode->groupId == COORDINATOR_GROUP_ID)
+	{
+		bool metadataSynced = true;
+		SetNodeMetadataSync(nodeName, nodePort, metadataSynced);
```

Comment:
I'm kind of wondering whether we should allow a state in which the hasmetadata, metadatasynced, or isactive flags are false for the coordinator at all.

Reply:
What would be the use-case for such a node?

Reply:
None, but we allow it today. So I kind of think we should not allow that at all.

Reply:
Do you mean not have the APIs to update (or throw errors when coordinator's these fields updated)?

Reply:
APIs should ensure the values are always true and disallow changing it. That way we don't create a problem we have to later fix.

Reply:
makes sense, applied your feedback. The commits/PR description should be updated, but I'll do it  before merging as we might still iterate on this code